src/argilla/server/daos/backend/search/model.py

```diff
@@ -74,13 +74,11 @@ class BaseRecordsQuery(BaseQuery):
 
     has_annotation: Optional[bool] = None
     has_prediction: Optional[bool] = None
-    # TODO(@ufukhurriyetogl): In order to allow embedding-based searches, we can include here
-    #  (the base records query data model class) an embeddings related query params that
+    # TODO(@ufukhurriyetoglu): In order to allow vector searches, we can include here
+    #  (the base records query data model class) a vectors related query params that
     #  we can use later for the records search.
     # The input vector:
-    embedding_vector: Optional[List[float]] = None
-    # The name of the embedding to uses (if several):
-    embedding_name: Optional[str] = None
+    # query: Optional[VectorQuery]
```

Comment:
not sure if I need to use the VectorQuery data model here, wdyt @frascuchon

Reply:
We should have a `VectorQuery` modeling the vector query parameters. But not should be the same class for the client and server. The client and server should have separate data models even if they share the same schema.

So, create here the new class and add it as a property in the query search model.